### PR TITLE
chore: move optimism bootnodes to `reth-primitives`

### PR DESCRIPTION
**Pull request by joshieDo**
As title suggests.

Additionally it fixes an issue where op stack chains were loading ethereum mainnet boot nodes (in addition to op ones) because there were no bootnodes associated with their chain specs.

### CODE REVIEW

File: crates/primitives/src/chain/spec.rs [hunk 1]
```diff
@@ -913,6 +916,14 @@ impl ChainSpec {
...
             C::Sepolia => Some(sepolia_nodes()),
             C::Holesky => Some(holesky_nodes()),
+            #[cfg(feature = "optimism")]
+            C::Base | C::Optimism => Some(op_nodes()),
+            #[cfg(feature = "optimism")]
+            C::BaseGoerli |
+            C::BaseSepolia |
+            C::OptimismGoerli |
+            C::OptimismSepolia |
+            C::OptimismKovan => Some(op_testnet_nodes()),
```
Comment (mattsse) [inline crates/primitives/src/chain/spec.rs:926]: is this correct?

this uses the same nodes for multiple different chains?

**Review by mattsse — APPROVED** [review]
lgtm, 1 q re match

File: crates/primitives/src/chain/spec.rs (hunk 1 above)
Comment (joshieDo) [reply crates/primitives/src/chain/spec.rs]: added the differentiation

**Review by mattsse — APPROVED** [review]

File: crates/net/network/src/config.rs [hunk 2]
```diff
@@ -132,14 +131,6 @@ impl<C> NetworkConfig<C> {
...
         }
 
-        if chain == Chain::optimism_mainnet() || chain == Chain::base_mainnet() {
-            builder = builder.add_optimism_mainnet_boot_nodes()
-        } else if chain == Chain::from_named(NamedChain::OptimismSepolia) ||
-            chain == Chain::from_named(NamedChain::BaseSepolia)
-        {
-            builder = builder.add_optimism_sepolia_boot_nodes()
-        }
-
```
Comment (emhane) [inline crates/net/network/src/config.rs:142]: how are you adding the boot nodes now that this is removed?

Comment (joshieDo) [reply crates/net/network/src/config.rs:142]: By adding the nodes to the ChainSpec, NetworkConfig will have them to serve to discovery_v5_with_config_builder

**Pull request merged**